README.md
# Tailwind CSS Highlight

![](https://raw.githubusercontent.com/ellreka/tailwindcss-highlight/main/assets/visual-image.png)

This extension highlights the utilities of TailwindCSS.

## Installation

<https://marketplace.visualstudio.com/items?itemName=ellreka.tailwindcss-highlight>

## Development

1. `yarn install`
2. `yarn dev`
3. Run debug mode on the vscode.

## Configuration

### `tailwindcss-highlight.languages`

List of languages to apply highlighting.

```json
"tailwindcss-highlight.languages": [
  "html",
  "javascript",
  "javascriptreact",
  "typescript",
  "typescriptreact",
  "vue",
  "php",
  "svelte"
]
```

### `tailwindcss-highlight.enabledUtilities`

List of utilities to highlight.

Utility name list: [src/defaultConfig.ts](https://github.com/ellreka/tailwindcss-highlight/blob/main/src/defaultConfig.ts)

```json
"tailwindcss-highlight.enabledUtilities": [
  "padding",
  "margin",
  "space",
  "width",
  "height",
  "flex",
  "grid",
  "position"
]
```

### `tailwindcss-highlight.customUtilitiesConfig`

Configuration for highlighting utilities.
Override [default config](https://github.com/ellreka/tailwindcss-highlight/blob/main/src/defaultConfig.ts).
Set only the properties you want to override.

| Name  | Description                                              | Type                                                                                                       |
| ----- | -------------------------------------------------------- | ---------------------------------------------------------------------------------------------------------- |
| regex | Regex to match the utility                               | `String`                                                                                                   |
| color | Color of the highlight. Style property takes precedence. | `String`                                                                                                   |
| style | Highlight styles                                         | [DecorationRenderOptions](https://code.visualstudio.com/api/references/vscode-api#DecorationRenderOptions) |

#### Example: Make it a underline instead of a highlight

```json
"tailwindcss-highlight.customUtilitiesConfig": {
  "padding": {
    "style": {
      "backgroundColor": "",
      "borderWidth": "0 0 3px 0",
      "borderStyle": "solid",
      "borderColor": "rgba(187, 196, 136, 1)",
      "color": "currentColor"
    }
  },
}
```

![](https://raw.githubusercontent.com/ellreka/tailwindcss-highlight/main/assets/example-configs.png)

### `tailwindcss-highlight.defaultVariantsColor`

Default color of the variants.

```json
"tailwindcss-highlight.defaultVariantsColor": "#1FAB89"
```

### `tailwindcss-highlight.customVariantsConfig`

Configuration for highlighting variants.

| Name  | Description     | Type       |
| ----- | --------------- | ---------- |
| Key   | Color           | `String`   |
| Value | List of variant | `String[]` |

```json
"tailwindcss-highlight.customVariantsConfig": {
  "#08D9D6": [
    "sm",
    "md",
    "lg",
    "xl",
    "2xl"
  ]
}
```

#### Example: Change the color of the focus variants

```json
"tailwindcss-highlight.customVariantsConfig": {
  "#FFB6C1": [
    "focus",
    "focus-within",
    "focus-visible"
  ]
}
```


CHANGELOG.md
# Change Log

## [v2.0.0](https://github.com/ellreka/tailwindcss-highlight/compare/v1.4.1...v2.0.0) - 2022-12-27

### Changed

- fix: update configuration by @ellreka in https://github.com/ellreka/tailwindcss-highlight/pull/93
  - Remove `tailwindcss-highlight.configs`, `tailwindcss-highlight.variants`.
  - Add `tailwindcss-highlight.enabledUtilities`, `tailwindcss-highlight.customUtilitiesConfig`, `tailwindcss-highlight.defaultVariantsColor`, `tailwindcss-highlight.customVariantsConfig`.
  - Change non-enabled utility style to dotted border.

## [v1.4.1](https://github.com/ellreka/tailwindcss-highlight/compare/v1.4.0...v1.4.1) - 2022-09-23

### Changed

- fix: fix vscode engines version by @ellreka in https://github.com/ellreka/tailwindcss-highlight/pull/77

## [v1.4.0](https://github.com/ellreka/tailwindcss-highlight/compare/v1.3.0...v1.4.0) - 2022-09-23

### Added

- feat: add variants highlighting by @ellreka in https://github.com/ellreka/tailwindcss-highlight/pull/64

### Changed

- ci: add tagpr by @ellreka in https://github.com/ellreka/tailwindcss-highlight/pull/62
- chore: update visual image and icon by @ellreka in https://github.com/ellreka/tailwindcss-highlight/pull/76

## Unreleased

- Highlight variants.

- Highlight the selected utility by the command.

## [1.3.0] 2022-07-20

## Changed

- Changed highlight style.

- Disabled some utilities.

## [1.2.8] 2022-07-03

## Fixed

- Fixed regex matching className.

## [1.2.6] 2021-04-16

## Changed

- Updated README.md.

## Fixed

- Apply changes to settings without reloading.

## [1.2.4] 2021-04-15

## Fixed

- Fixed some strings not being highlighted.

## [1.2.3] 2021-04-14

## Added

- Added `vue`, `php`, `svelte` to the default languages config.

## Fixed

- Fixed not working in jit mode.

## [1.2.1] 2021-05-26

## Changed

- Changed display name.

## [1.2.0] 2021-05-21

## Added

- Added some utilities.

- Added sample code for visual image.

## Fixed

- Delete utilities property from configuration.

- Fixed matching of strings other than the className.

## [1.1.4] 2021-05-19

## Fixed

- The build file was not included.

## [1.1.1] 2021-05-19

## Added

- Added some utilities.

- Added unit tests.

## Fixed

- Fixed bugs in regex.

## [1.0.1] - 2021-05-14

### Added

- Added keywords and icon into package.json.

## [1.0.0] - 2021-05-14

### Added

- Initial release 🎉


src/defaultConfig.ts
export const defaultEnabledUtilities = [
  'padding',
  'margin',
  'space',
  'width',
  'height',
  'flex',
  'grid',
  'position'
]

export const defaultUtilitiesConfig = {
  padding: {
    regex: '(?<=[:`\'"\\s])(|-)p(|t|b|r|l|x|y)-[^`\'"\\s]+',
    color: 'rgba(187, 196, 136, 1)'
  },
  margin: {
    regex: '(?<=[:`\'"\\s])(|-)m(|t|b|r|l|x|y)-[^`\'"\\s]+',
    color: 'rgba(173, 134, 91, 1)'
  },
  space: {
    regex: '(?<=[:`\'"\\s])(|-)space-(x|y)-[^`\'"\\s]+',
    color: 'rgba(173, 134, 91, 1)'
  },
  border: {
    regex: '(?<=[:`\'"\\s])border(?!-collapse|-separate)(|-)[^`\'"\\s]*',
    color: 'rgba(222, 195, 138, 1)'
  },
  rounded: {
    regex: '(?<=[:`\'"\\s])rounded(|-)[^`\'"\\s]*',
    color: 'rgba(222, 195, 138, 1)'
  },
  divide: {
    regex: '(?<=[:`\'"\\s])divide(|-)[^`\'"\\s]*',
    color: 'rgba(222, 195, 138, 1)'
  },
  ring: {
    regex: '(?<=[:`\'"\\s])ring(|-)[^`\'"\\s]*',
    color: 'rgba(222, 195, 138, 1)'
  },
  width: {
    regex: '(?<=[:`\'"\\s])(|-)(|max|min)(|-)w-[^`\'"\\s]+',
    color: 'rgba(160, 202, 222, 1)'
  },
  height: {
    regex: '(?<=[:`\'"\\s])(|-)(|max|min)(|-)h-[^`\'"\\s]+',
    color: 'rgba(230, 177, 161, 1)'
  },
  flex: {
    regex: '(?<=[:`\'"\\s])(|inline-)flex(|-)[^`\'"\\s]*',
    color: 'rgba(192, 175, 224, 1)'
  },
  grid: {
    regex: '(?<=[:`\'"\\s])grid(|-)[^`\'"\\s]*',
    color: 'rgba(192, 175, 224, 1)'
  },
  order: {
    regex: '(?<=[:`\'"\\s])(|-)order-[^`\'"\\s]+',
    color: 'rgba(192, 175, 224, 1)'
  },
  gridColumn: {
    regex: '(?<=[:`\'"\\s])col(|-)[^`\'"\\s]+',
    color: 'rgba(192, 175, 224, 1)'
  },
  gridRow: {
    regex: '(?<=[:`\'"\\s])row(|-)[^`\'"\\s]+',
    color: 'rgba(192, 175, 224, 1)'
  },
  gridAutoColumns: {
    regex: '(?<=[:`\'"\\s])auto-cols-[^`\'"\\s]+',
    color: 'rgba(192, 175, 224, 1)'
  },
  gridAutoRows: {
    regex: '(?<=[:`\'"\\s])auto-rows-[^`\'"\\s]+',
    color: 'rgba(192, 175, 224, 1)'
  },
  gap: {
    regex: '(?<=[:`\'"\\s])gap-[^`\'"\\s]+',
    color: 'rgba(192, 175, 224, 1)'
  },
  justifyContent: {
    regex: '(?<=[:`\'"\\s])justify-(start|end|center|between|around|evenly)',
    color: 'rgba(192, 175, 224, 1)'
  },
  justifyItems: {
    regex: '(?<=[:`\'"\\s])justify-items-(start|end|center|stretch)',
    color: 'rgba(192, 175, 224, 1)'
  },
  justifySelf: {
    regex: '(?<=[:`\'"\\s])justify-self-(auto|start|end|center|stretch)',
    color: 'rgba(192, 175, 224, 1)'
  },
  alignContent: {
    regex: '(?<=[:`\'"\\s])content-(start|end|center|between|around|evenly)',
    color: 'rgba(192, 175, 224, 1)'
  },
  alignItems: {
    regex: '(?<=[:`\'"\\s])items-(start|end|center|baseline|stretch)',
    color: 'rgba(192, 175, 224, 1)'
  },
  alignSelf: {
    regex: '(?<=[:`\'"\\s])self-(auto|start|end|center|stretch)',
    color: 'rgba(192, 175, 224, 1)'
  },
  placeContent: {
    regex:
      '(?<=[:`\'"\\s])place-content-(center|start|end|between|around|evenly|stretch)',
    color: 'rgba(192, 175, 224, 1)'
  },
  placeItems: {
    regex: '(?<=[:`\'"\\s])place-items-(start|end|center|stretch)',
    color: 'rgba(192, 175, 224, 1)'
  },
  placeSelf: {
    regex: '(?<=[:`\'"\\s])place-self-(auto|start|end|center|stretch)',
    color: 'rgba(192, 175, 224, 1)'
  },
  table: {
    regex: '(?<=[:`\'"\\s])(table-fixed|table-auto|table)',
    color: 'rgba(255, 210, 105, 1)'
  },
  borderCollapse: {
    regex: '(?<=[:`\'"\\s])border-(collapse|separate)',
    color: 'rgba(255, 210, 105, 1)'
  },
  transition: {
    regex: '(?<=[:`\'"\\s])transition(|-)[^`\'"\\s]*',
    color: 'rgba(141, 179, 184, 1)'
  },
  transitionDuration: {
    regex: '(?<=[:`\'"\\s])duration-[^`\'"\\s]*',
    color: 'rgba(141, 179, 184, 1)'
  },
  transitionTimingFunction: {
    regex: '(?<=[:`\'"\\s])ease(|-)[^`\'"\\s]*',
    color: 'rgba(141, 179, 184, 1)'
  },
  transitionDelay: {
    regex: '(?<=[:`\'"\\s])delay-[^`\'"\\s]*',
    color: 'rgba(141, 179, 184, 1)'
  },
  animation: {
    regex: '(?<=[:`\'"\\s])animate-[^`\'"\\s]+',
    color: 'rgba(255, 176, 202, 1)'
  },
  transform: {
    regex: '(?<=[:`\'"\\s])transform(|-)[^`\'"\\s]*',
    color: 'rgba(255, 176, 202, 1)'
  },
  transformOrigin: {
    regex: '(?<=[:`\'"\\s])origin-[^`\'"\\s]+',
    color: 'rgba(255, 176, 202, 1)'
  },
  scale: {
    regex: '(?<=[:`\'"\\s])scale-[^`\'"\\s]+',
    color: 'rgba(255, 176, 202, 1)'
  },
  rotate: {
    regex: '(?<=[:`\'"\\s])(|-)rotate-[^`\'"\\s]+',
    color: 'rgba(255, 176, 202, 1)'
  },
  translate: {
    regex: '(?<=[:`\'"\\s])(|-)translate-[^`\'"\\s]+',
    color: 'rgba(255, 176, 202, 1)'
  },
  skew: {
    regex: '(?<=[:`\'"\\s])(|-)skew-[^`\'"\\s]+',
    color: 'rgba(255, 176, 202, 1)'
  },
  text: {
    regex: '(?<=[:`\'"\\s])text-[^`\'"\\s]+',
    color: 'rgba(171, 169, 169, 1)'
  },
  textDecoration: {
    regex: '(?<=[:`\'"\\s])(underline|line-through|no-underline)',
    color: 'rgba(171, 169, 169, 1)'
  },
  textTransform: {
    regex: '(?<=[:`\'"\\s])(uppercase|lowercase|capitalize|normal-case)',
    color: 'rgba(171, 169, 169, 1)'
  },
  textOverflow: {
    regex: '(?<=[:`\'"\\s])(truncate|overflow-ellipsis|overflow-clip)',
    color: 'rgba(171, 169, 169, 1)'
  },
  verticalAlign: {
    regex: '(?<=[:`\'"\\s])align-[^`\'"\\s]+',
    color: 'rgba(171, 169, 169, 1)'
  },
  whitespace: {
    regex: '(?<=[:`\'"\\s])whitespace-(normal|nowrap|pre(-line|-wrap|))',
    color: 'rgba(171, 169, 169, 1)'
  },
  wordBreak: {
    regex: '(?<=[:`\'"\\s])break-(normal|words|all)',
    color: 'rgba(171, 169, 169, 1)'
  },
  font: {
    regex: '(?<=[:`\'"\\s])font-[^`\'"\\s]+',
    color: 'rgba(171, 169, 169, 1)'
  },
  fontStyle: {
    regex: '(?<=[:`\'"\\s])(italic|not-italic)',
    color: 'rgba(171, 169, 169, 1)'
  },
  leading: {
    regex: '(?<=[:`\'"\\s])leading-[^`\'"\\s]+',
    color: 'rgba(171, 169, 169, 1)'
  },
  tracking: {
    regex: '(?<=[:`\'"\\s])tracking-[^`\'"\\s]+',
    color: 'rgba(171, 169, 169, 1)'
  },
  background: {
    regex: '(?<=[:`\'"\\s])bg-[^`\'"\\s]+',
    color: 'rgba(117, 224, 165, 1)'
  },
  gradientColorStops: {
    regex: '(?<=[:`\'"\\s])(from|via|to)-[^`\'"\\s]+',
    color: 'rgba(117, 224, 165, 1)'
  },
  position: {
    regex: '(?<=[:`\'"\\s])(static|fixed|absolute|relative|sticky)',
    color: 'rgba(242, 142, 111, 1)'
  },
  inset: {
    regex: '(?<=[:`\'"\\s])(|-)(inset|top|bottom|left|right)-[^`\'"\\s]+',
    color: 'rgba(242, 142, 111, 1)'
  }
}
